### PR TITLE
Send DAG timeout callbacks to processor outside of prohibit_commit

### PR DESCRIPTION
**Pull request by tanelk**
Sending DAG timeout callbacks outside of prohibit_commit because when using standalone_dag_processor they are added to the database.

Issue reported in [slack](https://apache-airflow.slack.com/archives/CCQ7EGB1P/p1654788028424789)

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragement file, named `{pr_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/jobs/scheduler_job.py [hunk 1]
```diff
@@ -1159,10 +1159,7 @@ def _schedule_dag_run(
                 msg='timed_out',
             )
 
-            # Send SLA & DAG Success/Failure Callbacks to be executed
-            self._send_dag_callbacks_to_processor(dag, callback_to_execute)
```
Comment (ashb) [inline airflow/jobs/scheduler_job.py:1175]: Whoops, didn't this end up with the callback being sent twice before too?

Comment (ashb) [reply airflow/jobs/scheduler_job.py:1175]: Yet another case where our session handling causes bugs. Our use of `@create_session` which does an expunge_all _and_ closes the session when it exits needs fixing.

**Review by ashb — APPROVED** [review]

File: airflow/jobs/scheduler_job.py (hunk 1 above)
Comment (tanelk) [reply airflow/jobs/scheduler_job.py:1175]: No, the callback is executed once, because the `callback` it returned was `None`.

**Review by potiuk — APPROVED** [review]

**Comment by potiuk** [comment]
@mhenc  - can you also take a look here seems that standalone dag processor could cause problem with commit inside a "prohibit commit" guard.

**Pull request merged**